### ISSUE TITLE
Steming is giving too wide results

### ISSUE THREAD

**Issue by morchickit** [OpenDataServices/grantnav#508]
I queried GrantNav for "Clinical" and got results with the word clinic. I DID NOT want to get the word clinic at all... how can we make sure it doesn't make it too wide?

**Comment by robredpath** [comment]
The options are, I think:
- Export the data to another tool and use that tool to refine searches in cases where stemming has produced unhelpful results
- Try out other stemmers (see https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-stemmer-tokenfilter.html). It looks like the `light_english` stemmer might do what we want. But, of course, this would require some testing to see if it had its own issues. The stemmer wasn't a conscious choice AFAIK, but it's something I'd be cautious about changing)
- Write our own custom stemmer (probably overkill). 

Something we can consider as part of future GN work, for sure.

**Comment by mariongalley** [comment]
https://trello.com/c/tFLVz8eh/213-stemming-can-cause-confusing-or-inaccurate-results-in-grantnav

**Comment by mariongalley** [comment]
# Issue description:

When users are trying to make a specific search, elasticsearch will do fuzzy matching to look for similar words e.g. searching for "community organiser" will also return results containing "community organisation". This happens even when the user uses quotes, and when they use the AND operator, such as "community AND organiser".

This is an issue as "community organiser" and "community organisation" are very different things.

# Possible solutions:

- using quotes prevents fuzzy matching/synonyms being detected (I would have thought that the use of quotes would already return exact results, but apparently it does not)
- an operator which turns off fuzzy matching on the current search e.g. "EXACT community organiser"
- a toggle in the filters which allows you to choose between exact matching vs fuzzy matching

# It's clear how the world will be different when the work is completed

Users will have more control over whether GrantNav behaves more like Google i.e. "give me everything that might be related to the topic, no matter how tenuous" or more like a specialist search engine i.e. "give me all the items that exactly match my criteria"

**Comment by mariongalley** [comment]
@michaelwood In my latest comment I described a few possible solutions, all of which essentially boil down to having the option to turn off stemming/fuzzy matching. Is it possible to do this in GrantNav?

**Comment by michaelwood** [comment]
I think we could have an option to turn off the fuzzy matching. This would involve us having an internal field which isn't analysed  and could be switched to via the front end. 

That said I'd like to see the quotes method working, as you said this mirrors other known search paradigms and is expected behaviour. This may have been a regression when we changed version of ElasticSearch as I am fairly sure this used to work we may need now need to set the parameter https://www.elastic.co/guide/en/elasticsearch/reference/current/common-options.html#fuzziness based on whether there are quotes in the search string.

**Comment by mariongalley** [comment]
Can this be solved with guidance? If you search "Clinical" but don't want results with "Clinic" you can use the NOT operator and search "Clinical NOT Clinic"

**Comment by ruth360Giving** [comment]
We had another example of this in a recent intro to GrantNav workshop.  The user wanted to find grants for Animation but because of the stemming it was also returning grants for animals. Trying `animation NOT animals` returns zero results: https://grantnav.threesixtygiving.org/search?query=animation+NOT+animals&default_field=%2A&sort=_score+desc  Is there a way to stop it stemming between quotation marks maybe?

**Comment by mariongalley** [comment]
Another example @michaelwood if you search for "news" it returns "new" even with quotes

**Comment by michaelwood** [comment]
> We had another example of this in a recent intro to GrantNav workshop. The user wanted to find grants for Animation but because of the stemming it was also returning grants for animals. Trying `animation NOT animals` returns zero results: https://grantnav.threesixtygiving.org/search?query=animation+NOT+animals&default_field=%2A&sort=_score+desc 

> Is there a way to stop it stemming between quotation marks maybe?

Yes and no! As I understand it the elasticsearch mappings are the core of how this works and is similar to #1221 , what determines the results more than the actual query is how the data has been interpreted via the mappings (which include the analyser to use on a field). 

I think something has chanced in Elasticsearch as the current recommended approach is to map the field twice: once with a stemming analyser (for standard searching) and once with a non-stemming analyser (for quoted, exact phrase searches). The [quote_field_suffix ](https://www.elastic.co/docs/solutions/search/full-text/search-relevance/mixing-exact-search-with-stemming ) parameter then redirects any quoted query to the non-stemmed version of the field.

**Comment by mariongalley** [comment]
Thanks @michaelwood . I'd be happy to go ahead and try this approach.
